### DISCUSSION TITLE
保存 RSS 的全文到简悦的稍后读（包括快照）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3186] [服务]
> 通过简悦订阅 RSS 并将 RSS 的内容保存到本地快照。

# 使用场景

如果你有一个付费墙的 RSS，[在使用 Inoreader 加星保存到简悦](https://github.com/Kenshin/simpread/discussions/3007) 或  [将简悦变成你的 RSS 阅读器](https://github.com/Kenshin/simpread/discussions/3036)，会发现保存的快照并不是全文，而是部分内容。

这是因为上述方法都只是通过**获取原文来作为快照**的方案，当你的 RSS 为付费墙时，就无法做到获取原文了，有此问题的用户可以看下面的教程。

# 前置知识

> 需要事先完成下面的配置。

1. [如何开通开放平台](https://github.com/Kenshin/simpread/discussions/2757)
2. [API 快照功能](https://github.com/Kenshin/simpread/discussions/2994)
3. [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)

# integromat

> 如果你用过 IFTTT / Zapier 的话，integromat 是一个比 IFTTT 更强大的自动化程序，更难能可贵的是，免费版的即可使用 Webhook 服务。（ _Zapier 的 Webhook 收费，虽然 IFTTT 不收费，但它的定制化无法跟 integromat 相比_） 

> 此教程需要你事先注册 https://www.integromat.com/

# 如何配置

> 新建一个 scenario

![image](https://user-images.githubusercontent.com/81074/145666447-e37976f3-ed13-4311-be64-d39eb26510e8.png)

> 配置 RSS

![image](https://user-images.githubusercontent.com/81074/145663460-578fe8a8-4456-472c-9584-5364cc464e8e.png)

> 配置 HTTP（也就是你的 Webhook）

![image](https://user-images.githubusercontent.com/81074/145664654-eb793d19-dcbf-433b-bd6d-bd66143745d3.png)

**注意**

1. 你的 Webhook 地址 `https://api-wrap.simpread.pro/api/service/webhook/****?type=newsletter` 将 `*` 更换为你的 Token （登录开放平台后获取）

2. 上图最后一个图（三）的内容请复制下面的代码并粘贴过去。

3. 上图最后一个图（三）可能跟你的不一样，根据你的实际情况设置。一般来说你使用的 RSS 应该是输出全文的，请找到对应的全文。

```
{
    "url":"{{1.rssFields.link}}",
    "title":"{{1.rssFields.title}}",
    "data":"{{encodeURL(1.rssFields.description)}}"
}
```

> 确认上述无误后，即可完成配置。

# 如何测试

> RSS 的测试有些不同，请仔细留意视频中的操作。根据下图的测试，你的坚果云会马上收到一个 `.html` 文件，以及更新 `simpread_config.json` 通知。

https://user-images.githubusercontent.com/81074/145665086-92461c4f-a8a3-4386-b713-df49acc9fafd.mp4

# 开启

> RSS 的监控并不是实时监控，你可以根据实际情况取舍，比如我选择的就是 **每隔15分钟获取一次**

![image](https://user-images.githubusercontent.com/81074/145665941-bbb54ac9-1a4d-421d-9e52-7a39fcf6ef29.png)

# 最后

上面的步骤中，RSS 可更改为任意服务，如 Inoreader，但一定要保证下面两点：

1. 使用 `https://api-wrap.simpread.pro/api/service/webhook/****?type=newsletter` 将 `*` 更换为你的 Token （登录开放平台后获取）

2. 选择的时候，无论是否跟图示中的一样，但下图对应的内容跟你实际保持一致

    ![image](https://user-images.githubusercontent.com/81074/145666557-be201c05-1465-495e-b874-95c957b393ee.png)

    - `url` → 对应了地址
    - `title` → 对应了名称
    - `data` → 对应了全文

# 关联

- [利用 Airtable 实现属于自己的链接公开分享集 #3003](https://github.com/Kenshin/simpread/discussions/3003)
- [Raindrop.io 加星后自动导入到简悦并生成本地快照 #3021](https://github.com/Kenshin/simpread/discussions/3021)
- [加入 Pocket 后自动保存到简悦的稍后读（包含快照） #3024](https://github.com/Kenshin/simpread/discussions/3024)
- [Instapaper 加星后自动保存到简悦的稍后读（包含快照） #3034](https://github.com/Kenshin/simpread/discussions/3034)
- [将简悦变成你的 RSS 阅读器（自动下载快照到本地） #3036](https://github.com/Kenshin/simpread/discussions/3036)
- [保存到 Pinboard.in 同时也保存到简悦（包括本地快照） #3048](https://github.com/Kenshin/simpread/discussions/3048)
- [Inoreader 加星后自动保存到简悦的稍后读以及生成本地快照 #3007](https://github.com/Kenshin/simpread/discussions/3007)
- [当加入稍后读后，自动发送到你的 Telegram Channel #3092](https://github.com/Kenshin/simpread/discussions/3092)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）#3104](https://github.com/Kenshin/simpread/discussions/3104)

# 引申

- [[教程\]如何开通开放平台 #2757](https://github.com/Kenshin/simpread/discussions/2757)
- [当在开放平台设置完毕后，如何测试配置是否成功？ #3020](https://github.com/Kenshin/simpread/discussions/3020)
- [授权登录（如何使用第三方登录简悦的高级账户） #1570](https://github.com/Kenshin/simpread/discussions/1570)
- [个人中心 #2959](https://github.com/Kenshin/simpread/discussions/2959)
- [免密码登录 #2974](https://github.com/Kenshin/simpread/discussions/2974)

**Comment by YB-Kang** [comment]
![image](https://user-images.githubusercontent.com/55050981/162154355-06eadbf0-6365-43c0-a2c1-124a2c88750e.png)

请教老师，我根据配置教程来到测试部分，选择一篇订阅源文章后HTTP没反应，没有像上面测试的演示视频里那样出现个“1”，当然了稍后读也没增加内容。上面最后一个图那个设置我是照您给的代码直接复制进去的，这个需要做修改吗？

另外您说“上面的步骤中，RSS 可更改为任意服务，如 Inoreader，”，应该怎么理解？

**Reply by Kenshin** [reply]
重点是 [这个图](https://user-images.githubusercontent.com/81074/145664654-eb793d19-dcbf-433b-bd6d-bd66143745d3.png) 必须要在这里找到你选择的 RSS 源的具体内容。

每个 RSS 的字段不是 100% 一致。

**Reply by YB-Kang** [reply]
> 重点是 [这个图](https://user-images.githubusercontent.com/81074/145664654-eb793d19-dcbf-433b-bd6d-bd66143745d3.png) 必须要在这里找到你选择的 RSS 源的具体内容。
> 
> 每个 RSS 的字段不是 100% 一致。

理解了，问题解决了。我把您给的代码中的字段删掉，重新拖入就成功了。

**Reply by Kenshin** [reply]
哈哈，搞定就好。

**Reply by garyxj** [reply]
@Kenshin 保存成功，三个问题
1. 下载下来的快照html非全文
2. 稍后读内，看不到
3. 新的RSS，序号始终不变

**快照**
![image](https://user-images.githubusercontent.com/85046041/236167029-8aa34584-dfa6-4159-953e-fa081371f71e.png)


**稍后读**
![image](https://user-images.githubusercontent.com/85046041/236166766-a64324b3-1a95-491b-becf-85eafd425bc3.png)

**output 文件夹**
![image](https://user-images.githubusercontent.com/85046041/236166619-252e4ecd-e599-4eba-9210-8975ee177739.png)

**Reply by Kenshin** [reply]
@garyxj

> 下载下来的快照html非全文

这是由 RSS 源决定的，如果 RSS 源是全文，但本地快照不是全文的话，应该是配置错误。

> 稍后读内，看不到

造成这个问题的原因很多...

但一般来说，正确配置了知识库并且下载的 `.html` 是正确的，肯定会在稍后读中看到。

> 新的RSS，序号始终不变

有这个情况发生，一般说来是因为加入稍后读的，因为 API 走的网络，并非本地，所以可能坚果云客户端没有更新时，就同时保存了。
